cli/src/program.rs

```diff
@@ -2657,6 +2640,53 @@ fn do_process_program_upgrade(
     Ok(config.output_format.formatted_string(&program_id))
 }
 
+// Attempts to look up the program data account, and adds an extend program data instruction if the
+// program data account is too small.
+fn extend_program_data_if_needed(
+    initial_instructions: &mut Vec<Instruction>,
+    rpc_client: &RpcClient,
+    commitment: CommitmentConfig,
+    fee_payer: Pubkey,
```

Comment:
```suggestion
    fee_payer: &Pubkey,
```
Sorry, one more nit. Let's stay consistent with program_id. (While I'm sure there's a ton of inconsistency across the codebase, we do have a preference for passing pubkeys by reference.)

Reply:
Fixed.

I was wondering about it.
`Pubkey` is marked `Copy` and it is 32 bytes.
A pointer is 8 bytes, so we are adding an indirection and 25% more data %)
Why do we use references for `Pubkey`s?

Reply:
When `Pubkey` is passed by value, the compiler can pass it via a pointer:

https://godbolt.org/z/sso4eMx8Y

In this toy example, it actually treats `Pubkey` and `&Pubkey` as identical.

Reply:
> In this toy example, it actually treats `Pubkey` and `&Pubkey` as identical.

Then there should be no problem passing it by reference for consistency, eh? ;)

> Why do we use references for Pubkeys?

The goal was frankly to remove `Copy` for Pubkey. There are a lot of sensitive locations where it would matter. I understand it's rather meaningless here.